kml/types.hpp

```diff
@@ -344,27 +344,30 @@ struct TrackLayer
 struct MultiGeometry
 {
   using LineT = std::vector<geometry::PointWithAltitude>;
+  using TimeT = std::vector<double>;
```

Comment:
The original motivation was use doubles to workaround different precision on different platforms: some platforms provided seconds since the epoch, others provided milliseconds or microseconds. Using doubles solves this issue and makes processing timestamps in the core simpler and more precise.

Also see https://developer.apple.com/documentation/foundation/nsdate/1407504-timeintervalsince1970?language=objc

Any optimizations can be safely applied later, after making it work.